File: fizzbuzz.py
#!/usr/bin/env python2.7


def fizzbuzz(start, end):

    for i in range(start, end):

        mod5 = i % 5 == 0
        mod3 = i % 3 == 0

        if mod3 and mod5:
            print 'fizbuzz'
        elif mod5:
            print 'fizz'
        elif mod3:
            print 'buzz'


def main():
    fizzbuzz(1, 1000)

main()

File: fizzbuzz.rb
#!/usr/bin/ruby
start = ARGV[0].to_i
tail = ARGV[1].to_i

start.upto(tail) do |i|
    mod5 = i % 5 == 0
    mod3 = i % 3 == 0

    puts case
        when (mod3 and mod5) then 'fizzbuzz'
        when mod5 then 'fizz'
        when mod3 then 'buzz'
    end
end
